scripts/lib/CIME/XML/compilerblock.py

```diff
@@ -81,7 +81,7 @@ class CompilerBlock(object):
     matches_machine
     """
 
-    def __init__(self, writer, compiler_elem, machobj):
+    def __init__(self, writer, compiler_elem, machobj, db):
```

Comment:
Need to pass-in a db (some GenericXML object) so that we can use the generic_xml API.